Lesson23/Fırat/filter.js

```diff
@@ -172,248 +172,270 @@ console.log(usersOverTwentyFive);
+    quantity: 0,
+  },
+  {
+    name: "Gaming Console",
+    brand: "Microsoft",
+    price: 499,
+    quantity: 8,
+  },
+  {
+    name: "Tablet",
+    brand: "Samsung",
+    price: 649,
+    quantity: 18,
+  },
+  {
+    name: "Smart Speaker",
+    brand: "Amazon",
+    price: 99,
+    quantity: 0,
+  },
+  {
+    name: "Fitness Tracker",
+    brand: "Fitbit",
+    price: 129,
+    quantity: 14,
+  },
 ];
 
+const productsExercise6Result = productsExercise6.filter(
+  (product) => product.quantity === 0
+);
+console.log(productsExercise6Result);
+
 // Exercise 7:
 // Given an array of numbers, filter out the numbers that are divisible by 3.
 const numbersExercise7 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
 
+const numbersExercise7Result = numbersExercise7.filter(
+  (number) => number % 3 === 0
+);
+console.log(numbersExercise7Result);
+
 // Exercise 8:
 // Given an array of strings, filter out the strings that have more than 5 characters.
 const wordsExercise8 = [
-	'apple',
-	'banana',
-	'car',
-	'dog',
-	'elephant',
-	'flower',
-	'guitar',
-	'house',
-	'island',
-	'jungle',
+  "apple",
+  "banana",
+  "car",
+  "dog",
+  "elephant",
+  "flower",
+  "guitar",
+  "house",
+  "island",
+  "jungle",
 ];
 
+const wordsExercise8Result = wordsExercise8.filter((word) => word.length > 5);
+console.log(wordsExercise8Result);
+
 // Exercise 9:
 // Given an array of objects representing students, filter out the students who have a grade lower than C.
 const studentsExercise9 = [
-	{
-		id: 1,
-		name: 'Alice',
-		age: 18,
-		grade: 'A',
-	},
-	{
-		id: 2,
-		name: 'Bob',
-		age: 19,
-		grade: 'B',
-	},
-	{
-		id: 3,
-		name: 'Charlie',
-		age: 20,
-		grade: 'B+',
-	},
-	{
-		id: 4,
-		name: 'David',
-		age: 19,
-		grade: 'A-',
-	},
-	{
-		id: 5,
-		name: 'Emma',
-		age: 18,
-		grade: 'A+',
-	},
-	{
-		id: 6,
-		name: 'Frank',
-		age: 20,
-		grade: 'B',
-	},
-	{
-		id: 7,
-		name: 'Grace',
-		age: 19,
-		grade: 'D+',
-	},
-	{
-		id: 8,
-		name: 'Hannah',
-		age: 20,
-		grade: 'A-',
-	},
-	{
-		id: 9,
-		name: 'Isaac',
-		age: 18,
-		grade: 'D',
-	},
-	{
-		id: 10,
-		name: 'Julia',
-		age: 19,
-		grade: 'B-',
-	},
+  {
+    id: 1,
+    name: "Alice",
+    age: 18,
+    grade: "A",
+  },
+  {
+    id: 2,
+    name: "Bob",
+    age: 19,
+    grade: "B",
+  },
+  {
+    id: 3,
+    name: "Charlie",
+    age: 20,
+    grade: "B+",
+  },
+  {
+    id: 4,
+    name: "David",
+    age: 19,
+    grade: "A-",
+  },
+  {
+    id: 5,
+    name: "Emma",
+    age: 18,
+    grade: "A+",
+  },
+  {
+    id: 6,
+    name: "Frank",
+    age: 20,
+    grade: "B",
+  },
+  {
+    id: 7,
+    name: "Grace",
+    age: 19,
+    grade: "D+",
+  },
+  {
+    id: 8,
+    name: "Hannah",
+    age: 20,
+    grade: "A-",
+  },
+  {
+    id: 9,
+    name: "Isaac",
+    age: 18,
+    grade: "D",
+  },
+  {
+    id: 10,
+    name: "Julia",
+    age: 19,
+    grade: "B-",
+  },
 ];
 
+const studentsExercise9Result = studentsExercise9.filter(
+  (student) => student.grade >= "C"
```

Comment:
It would work well if you would compare numbers but with words it would not work as expected in all conditions.
Instead, you'd have compare the grade with every case one by one, e.g.
`student.grade !== "D+" || student.grade !== "D" || student.grade !== "D-"` and same for the F grade